internal/launcher/launcher.go

```diff
@@ -8,35 +8,44 @@ import (
 	"go.k6.io/k6/internal/build"
 )
 
-// Launcher is a k6 Launcher
-// It analyses the requirements of a k6 execution. If required, provisions a k6Runner to satisfy these
-// requirements.
+// binaryRunner executes the requested k6 command line command.
+// It abstract the execution path from the concrete binary.
+type binaryRunner interface {
+	run(*state.GlobalState)
+}
+
+// Launcher is a k6 launcher. It analyses the requirements of a k6 execution,
+// then if required, it provisions a binary executor to satisfy the requirements.
 type Launcher struct {
+	// gs is the global state of k6.
 	gs *state.GlobalState
-	// provision function receives a list of dependencies with their constrains and returns
-	// a k6runner than satisfies them
-	provision func(*state.GlobalState, k6deps.Dependencies) (k6Runner, error)
-	// k6Runner to execute k6 command
-	runner k6Runner
+
+	// provision generates a custom binary from the received list of dependencies
+	// with their constrains, and it returns an executor that satisfies them.
+	provision func(*state.GlobalState, k6deps.Dependencies) (binaryRunner, error)
+
+	// binaryRunner executes the requested k6 command line command
+	binaryRunner binaryRunner
```

Comment:
I'm not sure about this change. I prefer k6Runner as it reflects what it does and not how. Only the binary provisioning runner is using an external binary.

Reply:
I don't like the fact we have `k6` as a prefix because it doesn't add any value. It isn't clear what `k6` means in this context.

But I guess I've found one term that should address your request about expressing what it does.

Question: What it does?
Answer: It executes a sub-command. 

This is the reason why I propose `subcommandExecutor`. It integrates @joanlopez's feedback as well.

```suggestion
	subCmdExec subcommandExecutor
```

Reply:
I'm fine with `commandExecutor`. Going to submit the changes.